packages/google-vertex/src/google-vertex-auth-google-auth-library.ts

```diff
@@ -0,0 +1,27 @@
+import { GoogleAuth, GoogleAuthOptions } from 'google-auth-library';
+
+let authInstance: GoogleAuth | null = null;
+let authOptions: GoogleAuthOptions | null = null;
+
+function getAuth(options: GoogleAuthOptions) {
+  if (!authInstance || options !== authOptions) {
+    authInstance = new GoogleAuth({
+      scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+      ...options,
+    });
+    authOptions = options;
+  }
+  return authInstance;
+}
+
+export async function generateAuthToken(options?: GoogleAuthOptions) {
+  const auth = getAuth(options || {});
+  const client = await auth.getClient();
+  const token = await client.getAccessToken();
+  return token?.token || null;
+}
+
+// For testing purposes only
+export function _resetAuthInstance() {
+  authInstance = null;
+}
```

Comment:
Why is the singleton needed here?

Reply:
I was concerned (and briefly consulted claude) that the `GoogleAuth` instance could be a little heavyweight and/or something one wants to only construct once if possible. Looking briefly at the implementation I see it has a bunch of logic to cache things and allow parallel requests, which semi-supports the thinking.

https://github.com/googleapis/google-auth-library-nodejs/blob/08978822e1b7b5961f0e355df51d738e012be392/src/auth/googleauth.ts#L164

It could be premature optimization but it is a small and localized enough area where it felt worthwhile. Open to thought of course.

Reply:
FYI (and related to the cached Auth instance above in a way), in final testing (pack/deploy iterating) I noted the `node` route of `next-google-vertex` of course doesn't work as I haven't made a creds file available. I started looking into making one available to my deployment just to test fully, or learn more as to how folks commonly deploy this in Node, and found this:

https://googleapis.dev/nodejs/google-auth-library/9.14.1/

Under the **Loading credentials from environment variables** section:

> Instead of loading credentials from a key file, you can also provide them using an environment variable and the GoogleAuth.fromJSON() method. This is particularly convenient for systems that deploy directly from source control (Heroku, App Engine, etc).

I don't think adding optional support for this needs to block shipping, since:
- it doesn't appear there's a standard env var they use to trigger this
- I have had in mind that most folks ship a creds file if they're deploying this way (or use one of the other methods). One of the methods in fact tries to just "do the right thing", see **Choosing the correct credential type automatically**. I briefly tested this but it required enabling another GCP API (DNS related) and I haven't revisited yet.
- folks can implement this themselves with a custom headers hook
- we can wait to hear if there are requests and respond quickly with this or whatever else is requested